ui/src/message_popup/author/questions_page.jsx

```diff
@@ -59,18 +66,37 @@ export default React.createClass({
     return questionString.split(text).length;
   },
   
+  onQuestionsReceived(err, response){
+    const allDatabaseQuestions = JSON.parse(response.text).row;
+    if(allDatabaseQuestions === undefined) {
```

Comment:
I don't quite follow why the branching here is needed.  Maybe simplifying the shape of `getInitialState` would help?  We should be able to get the flow working where: `component mounts -> requests data -> gets data -> sets it in state` in one pass.